src/lib/sidenav/drawer.ts

```diff
@@ -385,13 +386,19 @@ export class MdDrawerContainer implements AfterContentInit, OnDestroy {
    * is properly hidden.
    */
   private _watchDrawerToggle(drawer: MdDrawer): void {
-    takeUntil.call(drawer._animationStarted, this._drawers.changes).subscribe(() => {
-      // Set the transition class on the container so that the animations occur. This should not
-      // be set initially because animations should only be triggered via a change in state.
-      this._renderer.addClass(this._element.nativeElement, 'mat-drawer-transition');
-      this._updateStyles();
-      this._changeDetectorRef.markForCheck();
-    });
+    RxChain.from(drawer._animationStarted)
+      .call(takeUntil, this._drawers.changes)
+      .call(filter, (event: AnimationEvent) => event.fromState !== event.toState)
+      .subscribe((event: AnimationEvent) => {
+        // Set the transition class on the container so that the animations occur. This should not
+        // be set initially because animations should only be triggered via a change in state.
+        if (event.toState !== 'open-instant') {
```

Comment:
```ts
this._element.nativeElement.classList.toggle('mat-drawer-transition', event.toState != 'open-instant');
```

Reply:
The second param isn’t supported on IE.

Reply:
Ah, in that case can we still at least use `classList.add` instead of the `Renderer`

Reply:
The renderer is SSR-friendly. AFAIK we can stop using it in Angular 5.

Reply:
Don't animation events not fire on the server?

Reply:
AFAIK they could fire if the sidenav starts off opened.

Reply:
Hmm ok, just needs rebase then